src/cpuid/src/common.rs

```diff
@@ -91,6 +92,39 @@ pub fn get_vendor_id_from_cpuid(cpuid: &CpuId) -> Result<[u8; 12], Error> {
     Err(Error::NotSupported)
 }
 
+/// Validates that a CPUID and another leaf 0x1 describe the same
+/// CPU in terms of manufacturer and model.
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+pub fn is_same_model(cpuid: &CpuId, other_leaf0x1_eax: u32) -> bool {
+    // Search for the entry for leaf0x1.
+    let leaf0x1 = cpuid
+        .as_slice()
+        .iter()
+        .find(|entry| entry.function == leaf_0x1::LEAF_NUM);
+    // Fail fast if we can't retrieve the relevant information from CPUID.
+    if leaf0x1.is_none() {
+        return false;
+    }
+
+    // The relevant information is in EAX.
+    let leaf0x1_eax = leaf0x1.unwrap().eax;
+
+    // Validate that all of these properties are the same.
+    for elem in &[
+        leaf_0x1::eax::EXTENDED_FAMILY_ID_BITRANGE,
+        leaf_0x1::eax::EXTENDED_PROCESSOR_MODEL_BITRANGE,
+        leaf_0x1::eax::PROCESSOR_FAMILY_BITRANGE,
+        leaf_0x1::eax::PROCESSOR_MODEL_BITRANGE,
+        leaf_0x1::eax::STEPPING_BITRANGE,
+    ] {
+        if leaf0x1_eax.read_bits_in_range(elem) != other_leaf0x1_eax.read_bits_in_range(elem) {
```

Comment:
Are this values guaranteed to be unique even across different vendors ? Shouldn't we also compare the vendor just to make sure ? The vendor can be extracted from the brand string and we might already have a helper method for that.

Reply:
My research led me to believe they are unique across every CPU model, which means different vendors would make them different as well.

However, the vendor string check doesn't hurt in this case so I added it just to be safe.